File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> s;
        vector<int> v(temperatures.size());
        int n = temperatures.size();
        for(int i=0; i<n; i++){
            while(!s.empty() && temperatures[s.top()] < temperatures[i]){
                v[s.top()] = i-s.top();
                s.pop();
            }
            s.push(i);
        }
        return v;
    }
};

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        nle = [0] * len(temperatures)
        stack = []
        for i in range(len(temperatures)-1, -1, -1):
            while(stack and temperatures[stack[-1]] <= temperatures[i]):
                stack.pop()
            if not stack:
                nle[i] = 0
            else:
                nle[i] = stack[-1] - i
            
            stack.append(i)
        
        return nle